### ISSUE TITLE
UID/GID does not work

### ISSUE THREAD

**Issue by b-chu** [Fullaxx/biglybt#4]
non-root user with custom group
Does not work - biglybt unable to connect to any trackers

**Comment by b-chu** [comment]
Fixable from #2 

```diff
diff Dockerfile
+ENV BIGLYBT_USER biglybt

+RUN groupadd -g 1000 ${BIGLYBT_USER} && \
+    useradd -rm -d /home/${BIGLYBT_USER} -s /bin/bash -u 1000 -g 1000 ${BIGLYBT_USER}

+    USER="${BIGLYBT_USER}" app_java_home="/usr/lib/jvm/java-11-openjdk-amd64/" /app/BiglyBT_Installer.sh -q && \
```


_Originally posted by @CWempe in https://github.com/Fullaxx/biglybt/issues/2#issuecomment-478773742_

**Comment by Fullaxx** [comment]
You are correct, and I believe the issue to be fixed now.
There was a typo in the bash script that was preventing permission changing.
Please let me know if you still experience issues using the new image.

**Comment by b-chu** [comment]
Wasn't fixed, same error. I think it's something with how you install biglybt
USER="${BIGLYBT_USER}" app_java_home="/usr/lib/jvm/java-11-openjdk-amd64/" /app/BiglyBT_Installer.sh

**Comment by Fullaxx** [comment]
can you post your docker run line?

**Comment by Fullaxx** [comment]
This is what I use and I can connect to trackers and download torrents:
docker run -d -p 127.0.0.1:5901:5901 -e VNCRES='1280x960' -e VNCUSER='guest' -e VNCUID='1000' fullaxx/biglybt

**Comment by b-chu** [comment]
VNCPASS="somestuff"
VNCUSER="box"
VNCUID="1000"
VNCGROUP="users"
VNCGID="1000"

**Comment by Fullaxx** [comment]
Thank you,
I see the issue now.
The users group already exists with GID of 100.
Try VNCGID="100"

I will look into patching it to read the GID from an existing group

**Comment by Fullaxx** [comment]
ok, thanks for the help. This appears to be fixed in the base image now.
If the VNCGROUP exists, Instead of bailing out it will use the existing GID number from the group file ignoring your VNCGID variable.
I have also updated the readme to be more specific.

Testing with docker run -d -p 127.0.0.1:5901:5901 -e VNCUSER="box" -e VNCUID="1000" -e VNCGROUP="users" -e VNCGID="1000" fullaxx/biglybt

Fix #4 with https://github.com/Fullaxx/ubuntu-desktop/commit/098189c8bd3c1d59d991219b10eb75357e0f1404

**Comment by b-chu** [comment]
VNCPASS="somestuff"
VNCUSER="biglybt"
VNCUID="1000"
VNCGROUP="biglybt"
VNCGID="1000"

still doesn't work

**Comment by Fullaxx** [comment]
Can you describe the error or post a screenshot? This is what I see when I run the command below.

tested with docker run -d -p 127.0.0.1:5901:5901 -e VNCPASS='stuff' -e VNCUSER='biglybt' -e VNCUID='1000' -e VNCGROUP='biglybt' -e VNCGID='1000' fullaxx/biglybt

![biglybt](https://user-images.githubusercontent.com/4896172/99272611-38de0300-27e5-11eb-872d-3aac5059af3b.png)

**Comment by b-chu** [comment]
Mine looks the same except under Status it says "Waiting"

**Comment by Fullaxx** [comment]
and everything works correctly, when you run it without user modification variables?
docker run -d -p 127.0.0.1:5901:5901 -e VNCPASS='stuff' fullaxx/biglybt

**Comment by b-chu** [comment]
Yep

**Comment by Fullaxx** [comment]
can you post a screenshot of an xterm window, like the one in the picture with the following commands:
id
cd /usr/local/biglybt
ls -l

I don't know how to solve a problem that I can't reproduce

**Comment by b-chu** [comment]
the output from mine looks exactly the same as the one you posted

It's fine I'll just use the workaround in #2

**Comment by Fullaxx** [comment]
sorry I couldn't be more help

**Comment by Fullaxx** [comment]
While I was never able to reproduce the specific "Waiting" issue described,
I am interested in creating an image that is usable by everyone equally.
I've been doing some testing into what might be causing that issue.

If anyone is still monitoring this thread, I would be interested to know if the latest published image solves the "Waiting" issue.
I did make some changes into how the software is installed that mimic the other fix described.

Tested with: docker run -d -p 127.0.0.1:5901:5901 -e VNCUSER="biglybt" -e VNCUID="1000" -e VNCGROUP="users" fullaxx/biglybt
